### PR TITLE
Move VTOrc tests to run against MySQL 8.0

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR moves the VTOrc tests to run against MySQL 8.0 instead of MySQL 5.7

## Checklist
- [x] Should this PR be backported? No
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

**Review by vmg — APPROVED** [review]
Strong :+1:. We need more stuff running in 8.0 because the 5.7 EOL is fast approaching.

**Comment by deepthi** [comment]
I'm conflicted about whether we can skip running this on 5.7 yet. How about running it on both?

**Comment by GuptaManan100** [comment]
I am conflicted too, at one hand we want to test both versions, on the other hand having tests for both versions is quickly going to go out of hand from the CI perspective. We only have 60 concurrent runners and expanding our test suite will reduce the velocity with which we can work.
I don't know what the best way forward is. I think the ideal situation would be have unit tests for these scenarios which we can run on both the versions of MySQL and then have the end to end tests run on either one version. WDYT?

**Comment by deepthi** [comment]
For now let us run this on both. This is the _only_ vtorc endtoend test, unlike other components where we have tons of endtoend tests and we could possibly choose to split them between 5.7 and 8.0.

**Comment by GuptaManan100** [comment]
@deepthi you got it. I'll make the changes

File: test/config.json [hunk 1]
```diff
@@ -1130,6 +1130,33 @@
...
+		},
+		"vtorc_general_8.0": {
+			"File": "unused.go",
+			"Args": ["vitess.io/vitess/go/test/endtoend/vtorc/general"],
+			"Command": [],
+			"Manual": false,
+			"Shard": "vtorc_8.0",
+			"RetryMax": 3,
+			"Tags": []
+		},
```
Comment (GuptaManan100) [inline test/config.json:1159]: I can avoid this duplication of same values of _8.0 by adding code to workflow generator where we remove `_8.0` suffix from the name of the test to find the shard to follow. This would mean that `vtorc_8.0` and `vtorc` will both have the command to follow the shard `vtorc` in the workflow and then we can remove this duplicate code.

I don't have a specific preference between the two, if you feel the other alternate is more readable/better, I'll make the change.

Comment (deepthi) [reply test/config.json:1159]: This is fine. Naming the test clearly helps in debugging when there are failures in CI.
